rclpy/rclpy/node.py

```diff
@@ -1652,3 +1652,68 @@ def assert_liveliness(self) -> None:
         """
         with self.handle as capsule:
             _rclpy.rclpy_assert_liveliness(capsule)
+
+    def _get_info_by_topic(
+        self,
+        topic_name: str,
+        no_mangle: bool,
+        func: Callable[[object, str, bool], List[Dict]]
+    ) -> List[Dict]:
+        fq_topic_name = expand_topic_name(topic_name, self.get_name(), self.get_namespace())
+        validate_topic_name(fq_topic_name)
```

Comment:
Should `validate_full_topic_name()` be used here? That's what the docs for `expand_topic_name()` suggest:

https://github.com/ros2/rclpy/blob/c108300f4068afbf6d839132a09f4dbf613b62b0/rclpy/rclpy/expand_topic_name.py#L24

Reply:
Switched over to using `validate_full_topic_name()`.

Note that `_count_publishers_or_subscribers()` had the same problem.